src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -43,6 +43,49 @@
 #endif
 
 
+///////////////////////////////////////////////////////////////////////
+/* SELFMADE -  Implementation of itoa() */
+
+char* myitoa(int num, char* str, int base) {
+  int i = 0;
+  bool isNegative = false;
+  /* Handle 0 explicitely, otherwise empty string is printed for 0 */
+  if (num == 0) {
+    str[i++] = '0';
+    str[i] = '\0';
+    return str;
+  }
+  // In standard itoa(), negative numbers are handled only with
+  // base 10. Otherwise numbers are considered unsigned.
+  if (num < 0 && base == 10) {
+    isNegative = true;
+    num = -num;
+  }
+  // Process individual digits
+  while (num != 0) {
+    int rem = num % base;
+    str[i++] = (rem > 9) ? (rem - 10) + 'a' : rem + '0';
+    num = num / base;
+  }
+
+  if (isNegative) str[i++] = '-';  // If number is negative, append '-'
+  str[i] = '\0';                   // Append string terminator
+  myreverse(str, i);               // Reverse the string
+  return str;
+}
+
+/* A utility function to reverse a string  */
+void myreverse(char str[], int length) {
+  for (int i = 0, j = length - 1; i < length / 2; i++, j--)  {
+    int temp = str[i];
```

Comment:
in str stehen doch chars und keine ints. könnte man auf `const unsigned char` casten.

Reply:
Ich vermute, du meinst `int temp = str[i];` welches du in `const unsigned char temp = str[i];` casten würdest?

Reply:
ja so hatte ich das gemeint. Richtig.

Reply:
Hier sind wir uns ja einig und diese Anpassung werde ich schon durchführen um Schritt für Schritt das Ergebnis umzusetzen.